src/pages/Reader.tsx

```diff
@@ -56,11 +58,40 @@ class Reader extends React.Component<any, State> {
         if (parsedSettings.menu) {
           readerSettings.menu = parsedSettings.menu;
         }
+        if (parsedSettings.autoLoadChapter) {
+          readerSettings.autoLoadChapter = parsedSettings.autoLoadChapter;
+        }
       }
       catch (err) {
         console.log('No settings found.')
       }
     }
+    const keyBindings = {
+      previousPage: "K",
+      nextPage: "J",
+      previousChapter: "H",
+      nextChapter: "L",
+    }
+    const keyBindingsLocal = localStorage.getItem('carrotKeyBindings')
+    if (keyBindingsLocal) {
+      try {
+        const keyBindingsParsed = JSON.parse(keyBindingsLocal);
```

Comment:
actually i still dont like how functions like these are handled.

a possible alternative is to define sm like
```
type Nullable<T> = { [K in keyof T]: T[K] | null };
```
and then do
```
const keyBindingsParsed: Nullable<Keybindings> = JSON.parse(keyBindingsLocal);
```
so now this makes sure that the localstorage object contains only the keys that are present in Keybindings. and then you can just object destructure
```
keybindings = {
  ...keybindingsDefault,
  ...keybindingsParsed
}
```

Reply:
I think I get what u mean, the nullable part is a bit confusing tho

Reply:
its defining a type where each property can be nullable, ex:
original type:
```
type MyType = {
  propOne: string,
  propTwo: boolean,
  propThree: number
}
```
then `Nullable<MyType>` will look like
```
type MyType = {
  propOne: string | null,
  propTwo: boolean | null,
  propThree: number | null
}
```

Reply:
Is null considered as undefined

Reply:
ah my bad, this should be undefined,
so
```
type Partial<T> = { [K in keyof T]: T[K] | undefined };
```
and naming it partial instead might make more sense, since it makes a version of the `T` object but each key is optional

Reply:
K